### ISSUE TITLE
Validations create empty translations

### ISSUE THREAD

**Issue by mrbrdo** [shioyama/mobility#569]
I am using fallbacks. One of my locales is "sl-SI", and I think the I18n fallbacks defaults to "sl" as a fallback.
However this bug exists also without fallbacks, it's just easier to explain this kind of case.

During validations, translation_for will be called:
https://github.com/shioyama/mobility/blob/master/lib/mobility/backends/active_record/table.rb#L291

It will build a translation for this "sl" locale, even though this is not necessary.
Additionally I have a presence validation on the Translation class (friendly_id slug). Therefore due to this, I cannot save my models, because fallbacks is generating empty translations on the model. Even if I did not have that validation, it would create unwanted empty translations in the DB upon saving.

**Comment by shioyama** [comment]
Please write a failing test, that's much clearer than a long explanation.

**Comment by mrbrdo** [comment]
See this failing spec: https://github.com/mrbrdo/mobility/blob/mrbrdo/spec/mobility/backends/active_record/table_spec.rb#L112

Also, I'm not sure that a fallback for `en-XX` should always include `en`, even if it's explicitly not specified (such as `fallbacks({ :'en-US' => 'de-DE', :pt => 'de-DE' })` will still include `en` and `de` fallbacks). This makes the above issue even worse, and I can imagine it causing problems in other situations too. I understand it from perspective of I18n with Yaml files, but it makes less sense for Mobility fallbacks, as if I am using `en-US` and not `en`, there is absolutely no reason I would have `en` translations in my DB.
I cannot add a failing spec for this part, because it is a discussion.

**Comment by mrbrdo** [comment]
@shioyama I think the fix could be as simple as https://github.com/shioyama/mobility/pull/572
I did some testing with my app (Spree) and it seems its working well. My failing spec is now passing with this change. Some existing specs need to be updated due to the change.

**Comment by dorner** [comment]
This is still a serious problem. I'd be willing to deal with cache misses if this would stop happening. Not only does it mess with validations (meaning I can't save records at all if I have a validation on the translation), it also caused a really difficult-to-pin-down bug when an `after_save` callback is defined which tries to read a value if it doesn't exist yet. I was trying to update the `updated_at` timestamp to the past, and it flat out wasn't working because the code destroys those dummy translations that they built and touches the original record to restore `updated_at` - *after* the save already happened. 

Long story short - I need this pretty badly :) I'll take a look at https://github.com/shioyama/mobility/pull/572 to see if I can get a better version working. @shioyama do you have any thoughts / suggestions for this? The case is really just "don't build a translation on read if it doesn't exist yet", which *should* be something we can patch for the table backend.

**Comment by phlegx** [comment]
We have a similar problem with `backend :table`, `active_record` and `dirty`. After disable dirty plugin with `dirty: false` on `translates` in the model, it works again.

Without disable dirty on the model, mobility correctly deletes nil translation (e.g. `en`) but then creates new translation (`en`) with  attribute nil.

Maybe it is related to `dirty` plugin?
